### ISSUE TITLE
Completing paths with spaces on windows

### ISSUE THREAD

**Issue by klemens** [kkawakam/rustyline#106]
Completing paths which include spaces does not work currently, as the space is part of the `DEFAULT_BREAK_CHARS` array and so `extract_word` does not extract the whole path.

A workaround is to [remove it from the list](https://github.com/klemens/rustyline/commit/2c492cc24a642b39dbbc58ef889edf69c651d59f) and making sure that you only pass the path without any prefix to the `FilenameCompleter`.

A proper fix would probably be doing what windows does: Adding quotes around the path if it contains a space.

**Comment by gwenn** [comment]
Currently the `escape`/`unescape` logic works only on unix.
Maybe we can try to use [shell-escape](https://crates.io/crates/shell-escape) (but the crate does not provide the `unescape`) !

**Comment by klemens** [comment]
Yes, using shell-escape seems better than trying to come up with our own logic. However, it always escapes using single quotes on unix, which might not be the desired behaviour (but it certainly is the simplest). Using it only for windows also seems possible, what do you think?

I guess `unescape` is missing, because it is only ever needed when actually writing a shell. But it shouldn't be too difficult to write a parser for quoted strings for windows. We also need to update `extract_word` to recognize these partially quoted strings correctly: `ls "my pict<TAB>`

I will see if I can come up with a pull request implementing this.
